CRM/Contact/BAO/Query.php

```diff
@@ -4924,7 +4924,10 @@ public function searchQuery(
    * @return CRM_Core_DAO
    */
   public function alphabetQuery() {
-    $query = $this->getSearchSQL(NULL, NULL, NULL, FALSE, FALSE, TRUE);
+    $sqlParts = $this->getSearchSQLParts(NULL, NULL, NULL, FALSE, FALSE, TRUE);
+    $sqlParts['group_by'] = 'GROUP BY sort_name';
+    $sqlParts['order_by'] = 'ORDER BY sort_name asc';
+    $query = "{$sqlParts['select']} {$sqlParts['from']} {$sqlParts['where']} {$sqlParts['having']} {$sqlParts['group_by']} {$sqlParts['order_by']}";
```

Comment:
May as well just write the GROUP BY & ORDER by straight into the query string?

Reply:
Also I think we should disableFullGroupBy in this function - it won't matter too much if we disable twice but it's clearer

Reply:
Also... looks like we can code in  SELECT 

    elseif ($sortByChar) {
      $select = 'SELECT DISTINCT LEFT(contact_a.sort_name, 1) as sort_name';
      $from = $this->_simpleFromClause;
    }

Reply:
@eileenmcnaughton I've coded in select and from directly.